File: Modul-6/Soal-5/PRAK605-2310817110013-Randy_Febrian.c
#include <stdio.h>

void input_matriks(int array, int matriks[array][array]) {
    for (int i = 0; i < array; i++) {
        for (int j = 0; j < array; j++) {
            scanf("%d", &matriks[i][j]);
        }
    }
}

int main() {
    int ordo;
    scanf("%d", &ordo);

    int matriks_1[ordo][ordo];
    int matriks_2[ordo][ordo];
    int hasil [ordo][ordo];
    printf("Matriks A\n");
    input_matriks(ordo, matriks_1);
    printf("Matriks B\n");
    input_matriks(ordo, matriks_2);

    printf("Matriks AXB\n");
    for (int i = 0; i < ordo; i++) {
        for (int j = 0; j < ordo; j++) {
            hasil[i][j] = 0;
            for (int k = 0; k < ordo; k++) {
                hasil[i][j] += matriks_1[i][k] * matriks_2[k][j];
            }
        }
    }
    
    for(int i = 0; i < ordo; i++){
        for (int j = 0; j < ordo; j++){
            printf("%d ", hasil[i][j]);
        }
        printf("\n");
    }
    return 0;
}

File: Modul-6/Soal-5/PRAK605-2310817110013-Randy_Febrian.py
import numpy as np

matriks_1 = []
matriks_2 = []
ordo = int(input())
print("Matriks A")
for i in range (ordo):
    matriks_1.append(np.array(list(map(int, input().split()))))
print("Matriks B")
for j in range (ordo):
    matriks_2.append(np.array(list(map(int, input().split()))))

matriks_1 = np.array(matriks_1).reshape(ordo, ordo)
matriks_2 = np.array(matriks_2).reshape(ordo, ordo)
hasil = np.dot(matriks_1, matriks_2)

print("Matriks AXB")
for row in hasil:
    for element in row:
        print(element, end=' ')
    print()